tests/test_skyWcs.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import lsst.utils.tests
+from lsst.afw.geom import SkyWcs, SpherePoint, Extent2D, Point2D, degrees, arcseconds, makeCdMatrix
+from lsst.afw.geom.testUtils import TransformTestBaseClass
+
+
+class TanSkyWcsTestCase(TransformTestBaseClass):
+    """Test pure TAN constructor of SkyWcs
+    """
+
+    def setUp(self):
+        TransformTestBaseClass.setUp(self)
+        self.crpix = Point2D(100, 100)
+        self.crval = SpherePoint(30 * degrees, 45 * degrees)
+        self.scale = 1.0 * arcseconds
+
+    def checkTanWcs(self, orientation, flipX):
+        """Construct a pure TAN SkyWcs and check that it operates as specified
+
+        Parameters
+        ----------
+        orientation : lsst.afw.geom.Angle
+            Position angle of focal plane +Y, measured from N through E.
+            At 0 degrees, +Y is along N and +X is along E/W if flipX false/true
+            At 90 degrees, +Y is along E and +X is along S/N if flipX false/true
+        flipX : bool
+            Fip x axis? See `orientation` for details.
+
+        Returns
+        -------
+        wcs : SkyWcs
+            The generated pure TAN SkyWcs
+        """
+        wcs = SkyWcs(
+            crpix = self.crpix,
+            crval = self.crval,
+            cdMatrix = makeCdMatrix(scale=self.scale, orientation=orientation, flipX=flipX),
+        )
+        self.checkPersistence(wcs)
+
+        xoffAng = 180*degrees if flipX else 0*degrees
+
+        pixelList = [
+            Point2D(self.crpix[0], self.crpix[1]),
+            Point2D(self.crpix[0] + 1, self.crpix[1]),
+            Point2D(self.crpix[0], self.crpix[1] + 1),
+        ]
+        skyList = wcs.tranForward(pixelList)
+
+        # check pixels to sky
+        predSkyList = [
+            self.crval,
+            self.crval.offset(xoffAng - orientation, self.scale),
+            self.crval.offset(90*degrees - orientation, self.scale)
+        ]
+        self.assertSpherePointListsAlmostEqual(predSkyList, skyList)
+        self.assertSpherePointListsAlmostEqual(predSkyList, wcs.pixelToSky(pixelList))
```

Comment:
SpherePoint tests look at angular separation on the sky and are only absolute, not relative. The default separation allowed is 0.001 arc seconds. It might be nice to expand the asserts to handle relative errors as well (feel free to file a ticket if you feel strongly about this).